TACC/Core-CMS
# GH-310: Breadcrumbs

# Overview

Add breadcrumbs to standard (a.k.a. Fullwidth template) pages.



# Issues

- GH-310



# Changes

- _Minor_: Update `trumps/README.md` to say some scopes enforce markup.
- __Major__: Add & Import styles.
- __Major__: Add & Load templates.



# Screenshots

| Description | Image |
| --- | --- |
| GH-310 (Full Width Template) | <img width="512" alt="GH-310 - Fullwidth Template" src="https://user-images.githubusercontent.com/62723358/129328892-78c3f1ac-6de4-4ed7-a5cd-1d71bef380e4.png"> | text is large†<br />no breadcrumbs✱ |
| GH-310 (Standard Template)<br />(with GH-98) | <img width="512" alt="GH-310 (with GH-98)" src="https://user-images.githubusercontent.com/62723358/129328886-57200028-1f78-4ccf-a39b-0a1e0b19c06b.png"> | text is small† |

<details><summary>Outdated</summary>

| Description | Image |
| --- | --- |
| GH-310 (Standard Template)<br />(sans GH-98) | <img width="512" alt="GH-310" src="https://user-images.githubusercontent.com/62723358/129328890-17abd696-eccf-42cd-a480-18fb3f32e44d.png"> | text is large† |

</details>

> ✱ Small text occurs with GH-98 code. Large text occurs without GH-98 code.
> † Because the fullwidth template is used by pages that do not expect breadcrumbs __and__ breadcrumbs are a new standard page feature __and__ all the fullwidth pages I have seen also want full height control (but I am not changing the template name, because that requires too much post-deploy work).




# Testing

1. If no environment, then [prepare to test CMS changes](https://github.com/TACC/Core-CMS/wiki/Test-CMS-Changes).
2. Create/Edit a page to use new Standard template.
    1. In `taccsite_cms/secrets.py`, to `_CMS_TEMPLATES`, add `('standard.html', 'Standard'),`.
    2. Choose "Standard" from Template in "Page" > "Templates" CMS admin menu.
3. Ensure breadcrumbs appear on page.
4. Ensure breadcrumbs and page content have padding to their left and right.
5. Ensure breadcrumb styles match the following criteria✱:
    1. The __links__, match those in [a relevant design](https://xd.adobe.com/view/90e6b252-0293-4243-7748-11d94a0763d0-e6d7/screen/ff0348ff-577d-450a-bef0-5326f3f7c705/specs/).
    2. The __character__ between entries is a __">"__ (not a "|").
    3. The __current page__ text is __not__ a link.
    4. The __current page__ text is bold text.
6. Set the page to not show in menu.
    - [How to Hide Page from Menu](https://user-images.githubusercontent.com/62723358/129304055-62885eae-8426-448b-b799-2c1817f91f82.png)
7. Ensure breadcrumbs __still__ appear on page.
8. Edit the page to use the Fullwidth template.
    1. Choose "Fullwidth" from Template in "Page" > "Templates" CMS admin menu.
9. Ensure breadcrumbs *do __not__† appear on page.

> ✱ Because they are wrapped in a "Container" (`<div class="container">`).
> † The font size may be ignored, because it is dependent on new global typography (upcoming PR).
